lms/djangoapps/discussion/signals/handlers.py

```diff
@@ -18,6 +20,25 @@
 ENABLE_FORUM_NOTIFICATIONS_FOR_SITE_KEY = 'enable_forum_notifications'
 
 
+@receiver(SignalHandler.course_published)
+def update_discussions_on_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
+    """
+    Catches the signal that a course has been published in the module
+    store and creates/updates the corresponding cache entry.
+    Ignores publish signals from content libraries.
+    """
+    if isinstance(course_key, LibraryLocator):
+        return
+
+    context = {
+        'course_id': unicode(course_key),
+    }
+    tasks.update_discussions_map.apply_async(
+        args=[context],
+        countdown=settings.DISCUSSION_SETTINGS['COURSE_PUBLISH_TASK_DELAY'],
```

Comment:
What's the reason for setting `countdown` here?

Reply:
Good question.  Here's the rationale from Block structures: https://github.com/edx/edx-platform/blob/03a3147634329fe989da253b1496624f2ff60096/lms/envs/common.py#L1933-L1937